### PR TITLE
feat: logs tables performance optimization

### LINKED ISSUE #3973: Log results section rebuild for performance and column reordering
### Which OpenObserve functionalities are relevant/related to the feature request?

_No response_

### Description

Current log result table grid layout uses quasar component which doesn't has column reordering feature. It takes time to render high volume of data. To improve this we need to consider light weight table component design which provide all these features.

### Proposed solution

We can consider external opensource library:
- https://datatables.net/

Let's discuss if any other opensource library provide table grid feature.

### Alternatives considered

NA

### PR DESCRIPTION
**Pull request by omkarK06**
fix #3973 

## Summary by CodeRabbit

- **New Features**
	- Introduced a new dynamic table component for enhanced data display and interaction.
	- Added support for Tailwind CSS, improving styling capabilities across the application.
	- Implemented interactive controls for managing log entries in the data table.

- **Improvements**
	- Enhanced logging functionality with clearer error messaging, improved content wrapping, and optimized selection of log fields based on user input.
	- Streamlined user interface in the JSON preview and search result components for better usability and readability.
	- Improved performance and responsiveness of the search results table.
	- Refined column visibility management in the search results, enhancing user experience.

- **Bug Fixes**
	- Updated column management logic in the search results for improved robustness.
	- Enhanced reliability of Playwright tests by refining timing and interaction steps.

### CODE REVIEW

**Review by bjp232004 — APPROVED** [review]

**Review by neha00290 — APPROVED** [review]

File: web/src/plugins/logs/data-table/CellActions.vue [hunk 1]
```diff
@@ -0,0 +1,69 @@
+<template>
+  <div
+    class="field_overlay tw-absolute tw-right-0 tw-top-[50%] table-cell-actions tw-translate-y-[-50%]"
+    :class="store.state.theme === 'dark' ? 'tw-bg-black' : 'tw-bg-white'"
```
[the remark this answers is not in the document]
Comment (omkarK06) [reply web/src/plugins/logs/data-table/CellActions.vue]: Fixed

File: web/src/plugins/logs/SearchResult.vue [hunk 2]
```diff
@@ -703,6 +435,22 @@ export default defineComponent({
...
+        (searchObj.config.splitterModel / 100);
+      return window.innerWidth - (leftSidebarMenu + fieldList) - 5;
+    });
+
+    watch(
+      () => searchObj.config.splitterModel,
+      (newValue, oldValue) => {
+        console.log(`splitterModel changed from ${oldValue} to ${newValue}`);
+      }
+    );
```
[the remark this answers is not in the document]
Comment (omkarK06) [reply web/src/plugins/logs/SearchResult.vue]: Removed watcher

File: web/src/plugins/logs/SearchResult.vue [hunk 3]
```diff
@@ -699,10 +432,26 @@
...
+        (searchObj.config.splitterModel / 100);
+      return window.innerWidth - (leftSidebarMenu + fieldList) - 5;
+    });
+
+    watch(
+      () => searchObj.config.splitterModel,
+      (newValue, oldValue) => {
+        console.log(`splitterModel changed from ${oldValue} to ${newValue}`);
+      },
+    );
```
[the remark this answers is not in the document]
Comment (omkarK06) [reply web/src/plugins/logs/SearchResult.vue]: Removed watcher

File: web/src/plugins/logs/SearchResult.vue [hunk 4]
```diff
@@ -459,6 +190,7 @@ import {
   onUpdated,
   onRenderTracked,
   defineAsyncComponent,
+  watch,
 } from "vue";
 import { copyToClipboard, useQuasar } from "quasar";
```
[the remark this answers is not in the document]
Comment (omkarK06) [reply web/src/plugins/logs/SearchResult.vue:212]: Removed console log

File: web/src/plugins/logs/SearchResult.vue [hunk 5]
```diff
@@ -699,10 +432,26 @@ export default defineComponent({
...
     };
 
+    const getTableWidth = computed(() => {
+      console.log("recalculate");
+      const leftSidebarMenu = 56;
+      const fieldList =
+        (window.innerWidth - leftSidebarMenu) *
+        (searchObj.config.splitterModel / 100);
+      return window.innerWidth - (leftSidebarMenu + fieldList) - 5;
+    });
```
[the remark this answers is not in the document]
Comment (omkarK06) [reply web/src/plugins/logs/SearchResult.vue]: Removed log statement

File: web/src/plugins/logs/JsonPreview.vue [hunk 6]
```diff
@@ -1,129 +1,126 @@
...
-        :name="'img:' + getImageURL('images/common/add_icon.svg')"
+        size="sm"
+        no-caps
+        class="q-px-sm"
+        icon="content_copy"
+        @click="copyLogToClipboard"
+      />
+    </div>
+    <div class="q-pl-md">
+      {
```
[the remark this answers is not in the document]
Comment (omkarK06) [reply web/src/plugins/logs/JsonPreview.vue:111]: This is required on UI, as you can see template has both closing bracket as well

File: web/src/plugins/logs/JsonPreview.vue [hunk 7]
```diff
@@ -1,129 +1,126 @@
...
+        <q-btn-dropdown
+          data-test="log-details-include-exclude-field-btn"
+          size="0.5rem"
+          flat
+          outlined
+          filled
+          dense
+          class="q-ml-sm pointer"
+          :name="'img:' + getImageURL('images/common/add_icon.svg')"
+        >
```
[the remark this answers is not in the document]
Comment (omkarK06) [reply web/src/plugins/logs/JsonPreview.vue]: Added

File: web/src/plugins/logs/JsonPreview.vue [hunk 8]
```diff
@@ -1,129 +1,126 @@
...
+            :data-test="`log-expand-detail-key-${key}-text`"
+            >{{ key }}:</span
+          ><span class="q-pl-xs" :data-test="`log-expand-detail-value-${value}`"
+            ><template v-if="index < Object.keys(value).length - 1"
+              >{{ value[key] }},</template
+            >
+            <template v-else>
+              {{ value[key] }}
+            </template>
+          </span>
```
[the remark this answers is not in the document]
Comment (omkarK06) [reply web/src/plugins/logs/JsonPreview.vue]: Fixed

File: web/src/plugins/logs/SearchBar.vue [hunk 9]
```diff
@@ -2176,6 +2194,7 @@ export default defineComponent({
         savedviewsService
           .put(store.state.selectedOrganization.identifier, viewID, viewObj)
           .then((res) => {
+            console.log(res,"res at after view updated  ")
             if (res.status == 200) {
               store.dispatch("setSavedViewDialog", false);
```
[the remark this answers is not in the document]
Comment (omkarK06) [reply web/src/plugins/logs/SearchBar.vue]: Removed

File: web/src/plugins/logs/SearchResult.vue [hunk 10]
```diff
@@ -750,6 +498,26 @@ export default defineComponent({
...
+
+    watch(
+      () => searchObj.config.splitterModel,
+      (newValue, oldValue) => {
+        console.log(`splitterModel changed from ${oldValue} to ${newValue}`);
+      },
```
[the remark this answers is not in the document]
Comment (omkarK06) [reply web/src/plugins/logs/SearchResult.vue]: Removed

**Pull request merged**